e2e/cypress/plugins/index.js

```diff
@@ -23,8 +23,12 @@ module.exports = (on, config) => {
     });
 
     if (process.env.NODE_ENV === 'qa') { // eslint-disable-line no-process-env
+        config.baseUrl = 'https://cypress.test.mattermost.com';
```

Comment:
I think this may overwrite an baseUrl we may pass into either via CLI or as Cypress_ENV variable. For example, in the pipeline we run cypress via `CYPRESS_baseUrl=https://cypress.test.mattermost.com npm run cypress:run`, so we have some redundant baseUrl assignment either in this new code or in the pipeline.

I think what we want is to know if we are running on CI or not, I propose something like:

```javascript
if (process.env.CI === true) { // eslint-disable-line no-process-env
    // If we are on CI, we need to use special urls for mailbox and webhooks
    config.mailboxUrl = `${config.baseUrl}/mail`;
    config.webhookBaseUrl = `${config.baseUrl}/webhook`;
} else {
    // Everything else would fall under running the app locally
    config.mailboxUrl = 'http://localhost:9000/api/v1/mailbox';
    config.webhookBaseUrl = 'http://localhost:3000';
}
```

I think this way we can pass in baseUrls normally (which can change or become unique once we parallelize). Note the change to `process.env.CI` I feel like that'd be a better choice of something to toggle on since it's fairly common in most CI tools for that to automatically get set to `true`.

Reply:
That's actually my intention; to pass ``NODE_ENV=qa`` when running cypress in CI.  In shared pipelines, I have prepared a [PR](https://github.com/mattermost/mattermost-shared-pipelines/pull/20/files) that do ``NODE_ENV=qa npx cypress run --spec="%/**/*";``

Another use case is for QA (Ogi and others) to be able to simply run E2E in their local machine against the test server just by running ``NODE_ENV=qa npx cypress run``

Honestly, I didn't consider yet parallel testing when I submitted this PR and never came to mind that we'll pass unique base URL for each instance.  I feel like when it comes, it should run against `http://localhost:8065` only (each parallel test on their own pod).

Let me think again.  I'll probably update per your suggestion.